shopinvader/models/shopinvader_variant.py

```diff
@@ -176,60 +176,108 @@ def _compute_variant_attributes(self):
+            # Get the price rule
+            price_unit, rule_id = pricelist.get_product_price_rule(product, qty, None)
+            # Get the price before applying the pricelist
+            SaleOrderLine = self.env["sale.order.line"].with_context(**product_context)
+            original_price_unit, currency = SaleOrderLine._get_real_price_currency(
+                product, rule_id, qty, product.uom_id, pricelist.id
             )
-            # fix tax on the real price
-            new_list_price = account_tax_obj._fix_tax_included_price_company(
-                new_list_price, product.taxes_id, tax_id, company
+            # Convert currency if necessary
+            if original_price_unit != 0:
+                if pricelist.currency_id != currency:
+                    original_price_unit = currency._convert(
+                        original_price_unit,
+                        pricelist.currency_id,
+                        company or self.env.company,
+                        fields.Date.today(),
+                    )
+            # Compute discount
+            price_dp = self.env["decimal.precision"].precision_get("Product Price")
+            if float_compare(
+                original_price_unit, price_unit, precision_digits=price_dp
+            ):
+                discount = (
+                    (original_price_unit - price_unit) / original_price_unit * 100
+                )
+                # Apply the right precision on discount
+                discount_dp = self.env["decimal.precision"].precision_get("Discount")
+                discount = float_round(discount, discount_dp)
+            else:
+                discount = 0.00
+            # Compute prices
+            original_price_unit = AccountTax._fix_tax_included_price_company(
+                original_price_unit, product.taxes_id, taxes, company
             )
-            product_precision = self.env["decimal.precision"].precision_get(
-                "Product Price"
+            res.update(
+                {
+                    "original_price": original_price_unit,
+                    "discount": discount,
+                }
             )
-            if (
-                float_compare(new_list_price, value, precision_digits=product_precision)
-                == 0
-            ):
-                # Both prices are equals. Product is wihout discount, avoid
-                # divide by 0 exception
-                return res
-            discount = (new_list_price - value) / new_list_price * 100
-            # apply the right precision on discount
-            dicount_precision = self.env["decimal.precision"].precision_get("Discount")
-            discount = float_round(discount, dicount_precision)
-            res.update({"original_value": new_list_price, "discount": discount})
+        # Keep backwards compatibility with deprecated keys
+        # TODO: Remove this in the future
+        res.update(
+            {
```

Comment:
is this change required? Can't we keep the old var name? This will just make the index size grow for no value.
Also, in most of the cases the new keys will be ignored by the frontend.

Reply:
Indeed, I'll revert this